woocommerce/woocommerce
# SelectTree: keep the focus on the input while navigating between menu items

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This increments the SelectTree component to maintain the focus on the input while navigating between menu items. Previously, the keyboard focus switched to the popover, which is not really a good practice from what I see everywhere. It also tweaks some ARIA attributes based on my tests with screen reader.

This is meant to be a short term improvement while we discuss next steps for this field, possibly creating something upstream to core.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
2. Go to Products > Add
3. Go to Organization tab
4. Go to the Categories field
5. Test the field: focus on the input and select categories using the up/down/left/right arrow keys and Enter to select. The focus should remain on the input and you should be able to type to filter at any point.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
